geopandas/array.py

```diff
@@ -30,6 +30,12 @@
 from .sindex import _get_sindex_class
 
 
+class class_or_instancemethod(classmethod):
+    def __get__(self, instance, type_):
```

Comment:
Can you add a short docstring explaining what is going on and maybe linking back to this PR for further context?